test/integration/tcproute_test.go

```diff
@@ -432,6 +433,275 @@ func TestTCPRouteEssentials(t *testing.T) {
 	}, ingressWait, waitTick)
 }
 
+func TestTCPRouteReferencePolicy(t *testing.T) {
+	ns, cleanup := namespace(t)
+	defer cleanup()
+	t.Log("locking TCP port")
+	tcpMutex.Lock()
+	defer func() {
+		t.Log("unlocking TCP port")
+		tcpMutex.Unlock()
+	}()
+
+	other, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name()+"other")
```

Comment:
The `other` variable is pretty generic, perhaps:

```suggestion
	extraNamespace, err := clusters.GenerateNamespace(ctx, env.Cluster(), uuid.NewString())
```